azurerm/data_source_public_ip_prefix.go

```diff
@@ -0,0 +1,65 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmPublicIpPrefix() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmPublicIpPrefixRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"prefix_length": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"ip_prefix": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"zones": azure.SchemaSingleZone(),
```

Comment:
since this is a Data Source these are ReadOnly - as such could we make this:

```suggestion
			"zones": azure.SchemaZonesComputed(),
```

Reply:
It looks like the difference is `SchemaSingleZone` has:
- `ForceNew: true`
- `MaxItems: 1`

And `SchemaZonesComputed` has:
`Computed: true`

I'm not sure how to know when to use which one.
(Maybe since there is a `ForceNew`, it shouldn't be used for data sources).